Game4.py

```diff
@@ -0,0 +1,165 @@
+import os, sys
+import pygame
+from random import randint
+
+from misc import *
+
+
+class Game4Main:
+    """The Main Game4 Class - This class handles the main
+    initialization and creating of the Game."""
+
+    def __init__(self, width=480,height=360):
+        # initialize pygame
+        pygame.init()
+        # set window size
+        self.width = width
+        self.height = height
+        # create screen
+        self.screen = pygame.display.set_mode((self.width, self.height))
+        # create clock
+        self.clock = pygame.time.Clock()
+        self.score = 0
+
+
+    def mainLoop(self):
+        """Main screen for game."""
+        # initialize player
+        count = 0
+        play_len = 25
+        play_x = 100
+        play_y = self.height - play_len
+        player = Player(play_x, play_y, play_len, self.screen)
+
+        # initialize stamina bars
+        P1_stamina_bar = StaminaBar(self.screen,25,"WHITE")
+        P2_stamina_bar = StaminaBar(self.screen,350,"RED")
+
+        # initialize obstacle length, x and y coordinates
+        obs_len = 25
+        obs_x = self.width
+        obs_y = self.height - obs_len
+        obstical_height = self.height - obs_len
+
+        # create list of obstacles
+        obstacles = []
+        new_obstical = False
+        # uncomment and place line below in [] of obstacles = [] to generate random obstacles:
+        # Obstacle(obs_x, obs_y, obs_len, self.screen,'BLUE')
+
+
+        # initialize time variables
+        prev_time = 0
+        obs_dt = 500
+
+        # main event loop
+        while 1:
```

Comment:
`while 1` does not provide any direct information about when the loop is supposed to end. Using a boolean variable named `running` provides better documentation in this case.

Reply:
Yes, prefer `True` or `False` for a boolean.

Going beyond: you can also use [itertools](https://docs.python.org/3.6/library/itertools.html#itertools.count) to produce a continuously increasing set of integers, like `range(1, 100)` except it keeps going forever instead of stopping at 100:

`for count in itertools.count(1):`